numpy/core/src/multiarray/datetime.c

```diff
@@ -1761,6 +1762,9 @@ parse_datetime_unit_from_string(char *str, Py_ssize_t len, char *metastr)
                 return NPY_FR_as;
         }
     }
+    else if (allow_generic && len == 7 && !strncmp(str, "generic", 7)) {
```

Comment:
Perhaps it would be simpler to tell me your preference here. I do not feel qualified to make design decisions for the Numpy core team.